server/core/src/storage.ts

```diff
@@ -337,6 +337,25 @@ class TServerStorage implements ServerStorage {
     return result
   }
 
+  async processMove (ctx: MeasureContext, tx: Tx): Promise<Tx[]> {
+    const actualTx = this.extractTx(tx)
+    if (!this.hierarchy.isDerived(actualTx._class, core.class.TxUpdateDoc)) return []
+    const rtx = actualTx as TxUpdateDoc<Doc>
+    if (rtx.operations.space === undefined || rtx.operations.space === rtx.objectSpace) return []
+    const result: Tx[] = []
+    const factory = new TxFactory(core.account.System)
+    for (const [, attribute] of this.hierarchy.getAllAttributes(rtx.objectClass)) {
+      if (!this.hierarchy.isDerived(attribute.type._class, core.class.Collection)) continue
+      const collection = attribute.type as Collection<AttachedDoc>
+      const allAttached = await this.findAll(ctx, collection.of, { attachedTo: rtx.objectId, space: rtx.objectSpace })
```

Comment:
We need to call refactor buildRemovedDoc to allow find previous space value for document, and find only attached documents with it's space value.

Reply:
Discussed, resolved